go/cmd/vtctldclient/command/transactions.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"vitess.io/vitess/go/cmd/vtctldclient/cli"
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	vtctldatapb "vitess.io/vitess/go/vt/proto/vtctldata"
+)
+
+var (
+	DistributedTransaction = &cobra.Command{
+		Use:   "DistributedTransaction <cmd>",
+		Short: "Perform commands on distributed transaction",
+		Args:  cobra.MinimumNArgs(2),
+
+		DisableFlagsInUseLine: true,
+	}
+
+	// GetUnresolvedTransactions makes an GetUnresolvedTransactions gRPC call to a vtctld.
+	GetUnresolvedTransactions = &cobra.Command{
+		Use:     "list <keyspace>",
+		Short:   "Retrieves unresolved transactions for the given keyspace.",
+		Aliases: []string{"List"},
+		Args:    cobra.ExactArgs(1),
+		RunE:    commandGetUnresolvedTransactions,
+
+		DisableFlagsInUseLine: true,
+	}
+
+	// ConcludeTransaction makes a ConcludeTransaction gRPC call to a vtctld.
+	ConcludeTransaction = &cobra.Command{
+		Use:     "conclude <dtid> [<keyspace/shard> ...]",
+		Short:   "Concludes the unresolved transaction by rolling back the prepared transaction on all participating shards and removing the transaction metadata record.",
+		Aliases: []string{"Conclude"},
+		Args:    cobra.MinimumNArgs(1),
```

Comment:
From the examples it is not entirely clear to me what the arguments to conclude will look like. But we have found it more useful to specify the parameters as flags rather than depend on `args` because it is easy to forget the order or delimiters etc. We had that problem a lot with the earlier vreplication commands. It is a bit more typing

For example it could be 
`DistributedTransactions Conclude --keyspace ks1 --shards -40,a0- --dtid 82345` OR with shorthand codes

`DistributedTransactions Conclude -k ks1 -s -40,a0- --dtid 82345` 

You can decide if this works better or not ...

Reply:
A transaction can span multiple keyspaces so keyspace and shard will go in pair like `keyspace/shard`

Reply:
I did not find a good way to record the participants list. 
We keep it like this for now.